src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EmbeddedLanguages/Common/EmbeddedSyntaxNode.cs

```diff
@@ -103,6 +105,54 @@ public bool Contains(VirtualChar virtualChar)
             return false;
         }
 
+        /// <summary>
+        /// Returns the string representation of this node, not including its leading and trailing trivia.
+        /// </summary>
+        /// <returns>The string representation of this node, not including its leading and trailing trivia.</returns>
+        /// <remarks>The length of the returned string is always the same as Span.Length</remarks>
+        public override string ToString()
+        {
+            using var _ = PooledStringBuilder.GetInstance(out var sb);
+            WriteTo(sb, leading: false, trailing: false);
+            return sb.ToString();
+        }
+
+        /// <summary>
+        /// Returns full string representation of this node including its leading and trailing trivia.
+        /// </summary>
+        /// <returns>The full string representation of this node including its leading and trailing trivia.</returns>
+        /// <remarks>The length of the returned string is always the same as FullSpan.Length</remarks>
+        public string ToFullString()
+        {
+            using var _ = PooledStringBuilder.GetInstance(out var sb);
+            WriteTo(sb, leading: true, trailing: true);
+            return sb.ToString();
+        }
+
+        /// <summary>
+        /// Writes the node to a stringbuilder.
+        /// </summary>
+        /// <param name="leading">If false, leading trivia will not be added</param>
+        /// <param name="trailing">If false, trailing trivia will not be added</param>
+        public void WriteTo(StringBuilder sb, bool leading, bool trailing)
+        {
+            for (var i = 0; i < this.ChildCount; i++)
+            {
+                var child = this[i];
+                var currentLeading = leading || i > 0;
+                var curentTrailing = trailing || i < (this.ChildCount - 1);
```

Comment:
something about this feels off.  we should only be 'leading' if we were leading and we're only on the first child.  teh `i>0` makes it so the following children will be considered leading as well.  

Similarly with trailing, only teh last child of the current trailing item is also trailing.

Reply:
oh/n'm i get what htis is saying.